forte/utils/payload_factory.py

```diff
@@ -0,0 +1,159 @@
+
+        Args:
+            meta_name: a Generic object that is used to register a Payload meta data type.
+        """
+        if meta.source_type not in ("web", "local"):
+            raise ValueError("Meta data source must be either 'web' or 'local'")
+        self.valid_meta[type(meta)] = True
+
+    def check_meta(self, meta):
+        return type(meta) in self.valid_meta
+
+
+class Payloading:
+    """
+    An class that help mapping meta data to loading function.
+    """
+
+    def __init__(self):
+        self._factory = PayloadFactory()
+
+    def load_factory(self, factory):
+        self._factory = factory
+
+    @abstractmethod
+    def route(self, meta):
+        """
+        Convert the meta into a loading function that takes uri and read data
+        from the uri.
+
+        Args:
+            meta: Meta data
+
+        Returns:
+            a function that takes uri and read data from the uri.
+        """
+        pass
+
+
+class ImagePayloading(Payloading):
+    """
+    A class that helps mapping Image meta data to loading function.
+    """
+
+    def route(self, meta) -> Callable:
+        """
+        A function that takes and analyzes a meta data and returns a
+        corresponding loading function.
+
+        Args:
+            meta: a Meta object that is used to determine the loading function.
+
+        Raises:
+            ValueError: if meta is not a valid meta data.
+
+        Returns:
+            a function that takes uri and read data from the uri, and it returns the data.
+        """
+        if not self._factory.check_meta(meta):
+            raise ValueError(f"Meta data{meta} not supported")
+        if not meta.source_type in ("web", "local"):
+            raise ValueError("Meta data source must be either 'web' or 'local'")
+
+        if meta.source_type == "local":
+            try:
+                import matplotlib.pyplot as plt  # pylint: disable=import-outside-toplevel
+            except ModuleNotFoundError as e:
+                raise ModuleNotFoundError(
+                    "ImagePayloading reading local file requires `matplotlib`"
+                    "package to be installed."
+                ) from e
+            return plt.imread
+        else:
+            try:
+                from PIL import Image  # pylint: disable=import-outside-toplevel
+                import requests  # pylint: disable=import-outside-toplevel
+            except ModuleNotFoundError as e:
+                raise ModuleNotFoundError(
+                    "ImagePayloading reading web file requires `PIL` and"
+                    "`requests` packages to be installed."
+                ) from e
+
+            def read_uri(uri):
+                # customize this function to read data from uri
+                uri_obj = requests.get(uri, stream=True)
+                pil_image = Image.open(uri_obj.raw)
+                return self._pil_to_nparray(pil_image)
```

Comment:
You need to provide a way for user to customize it, now this is buried deeply in our code